forte/helpers/lbfgs/lbfgs.cc

```diff
@@ -420,14 +415,14 @@ double LBFGS::compute_gamma() {
     double value = 1.0;
     if (iter_ - iter_shift_) {
         int end = (iter_ - iter_shift_ - 1) % (std::min(iter_ - iter_shift_, param_->m));
-        value = s_[end]->vector_dot(y_[end]) / y_[end]->vector_dot(y_[end]);
+        value = s_[end]->vector_dot(*y_[end]) / y_[end]->vector_dot(*y_[end]);
     }
     return value;
 }
 
 void LBFGS::resize(int m) {
-    y_ = std::vector<psi::SharedVector>(m, std::make_shared<psi::Vector>());
-    s_ = std::vector<psi::SharedVector>(m, std::make_shared<psi::Vector>());
+    y_ = std::vector<psi::SharedVector>(m, std::make_shared<psi::Vector>(0));
+    s_ = std::vector<psi::SharedVector>(m, std::make_shared<psi::Vector>(0));
```

Comment:
I think the following is valid.
```suggestion
    y_ = std::vector<psi::SharedVector>(m, std::make_shared<psi::Vector>(m));
    s_ = std::vector<psi::SharedVector>(m, std::make_shared<psi::Vector>(m));
```

Reply:
Just to confirm, expected/desirable behavior is to create a vector of `m` SharedVectors with `m` elements? This would be a behavior change.